File: all-code/101-200/190reverseBits.py

class Solution:
    def reverseBits(self, n: int) -> int:
        res, cnt = 0, 0
        while cnt < 32:
            res = (res << 1) + (n & 1)
            n >>= 1
            cnt += 1
        return (res)

so = Solution()

print(so.reverseBits(10))
print(so.reverseBits(43261596))
print(so.reverseBits(4294967293))

File: c++/src/1-999/190reverseBits.cpp
#include "lc_pub.h"

class Solution {
    
    public:
    int reverseBits(int n) {
        int ans=0;
        for (int i=0;i<32;i++) {
            ans = (ans << 1) + (n & 1);
            n >>= 1;
        }
        return ans;
    }
};

int main()
{
    std::cout<<"test let us start! %s" << __cplusplus <<std::endl;

    Solution so;
    cout<< so.reverseBits(3)<<endl;
    return 0;
}
